### PR TITLE
Jaskowicz1/sslclient changes

### PR DESCRIPTION
**Pull request by Neko-Life**
## Changes in this PR

- Refactor http_version and fix version number should not be hex number
- Fix seeking without checking for file value causing segfault in unittest
- ~~Add hex_to_str utility to convert hex to string number~~

## Code change checklist

- [x] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [x] My code follows the [coding style guide](https://dpp.dev/coding-standards.html).
- [x] I tested that my change works before raising the PR.
- [x] I have ensured that I did not break any existing API calls.
- [x] I have not built my pull request using AI, a static analysis tool or similar without any human oversight.

### CODE REVIEW

**Comment by Jaskowicz1** [comment]
I'll merge this up and change the hex_to_str to be a one liner, other than that, all gooddd.

File: src/unittest/test.h [hunk 1]
```diff
@@ -192,6 +192,7 @@ DPP_TEST(UTILITY_CHANNEL_URL, "utility::channel_url", tf_offline);
 DPP_TEST(UTILITY_THREAD_URL, "utility::thread_url", tf_offline);
 DPP_TEST(UTILITY_AVATAR_SIZE, "utility::avatar_size", tf_offline);
 DPP_TEST(UTILITY_CDN_ENDPOINT_URL_HASH, "utility::cdn_endpoint_url_hash", tf_offline);
+
```
Comment (Jaskowicz1) [inline src/unittest/test.h:195]: empty line needs removing

File: src/unittest/test.cpp [hunk 2]
```diff
@@ -613,6 +619,7 @@ Markdown lol \\|\\|spoiler\\|\\| \\~\\~strikethrough\\~\\~ \\`small \\*code\\* b
 				 dpp::utility::thread_url(1,0) == "" &&
 				 dpp::utility::thread_url(0,0) == "" 
 		);
+
```
Comment (Jaskowicz1) [inline src/unittest/test.cpp:622]: empty line needs removing here too :)

**Review by Jaskowicz1 — CHANGES_REQUESTED** [review]

**Comment by Neko-Life** [comment]
hold on renaming my branch

**Comment by Jaskowicz1** [comment]
Panicked there hahahaha

**Pull request closed without merging**